aiida_quantumespresso/calculations/dos.py

```diff
@@ -1,20 +1,13 @@
 # -*- coding: utf-8 -*-
 from __future__ import absolute_import
-from aiida.orm import RemoteData, FolderData, CalcJobNode
-from aiida.orm import Dict, XyData
+
+from aiida import orm
```

Comment:
Is this just your preference or do you want to enforce this style across the code? Because I'm not a fan of it.

Reply:
It is a personal preference yes, for the following reasons:
* It is less sensitive to internal path changes in the `aiida.orm` module
* It makes the import section less cluttered
* Adding code with new import requirements do not require new import statements
* Don't run the risk of forgetting to remove unused imports when removing code.

The one counter-argument could be that importing the whole module will be slower, but it is not as `aiida.orm` already import all lower lying modules anyway. At 30 - 60 micro seconds I anyway do not think that this weighs up to the advantages listed above.

Reply:
Good points, however the one counter-argument to me is that now you have cluttered all mentions of orm classes with a `orm.` prefix, and to me there's no need since it's clear enough (I believe) that they belong to AiiDA already.

Reply:
Unused imports and missing imports should be caught by the linter. And I don't get why it's less sensitive to internal path changes, as you will have to use the rest of path anyway when you use the class... Or not?

Reply:
The import robustness comes into play when importing like `from aiida.orm.nodes.data.str import Str`, which becomes more likely if we have multiple individual import statements. This would have to be changed if the internal path of `aiida-core` changes. For `v1.0.0` we have [specified a list](https://github.com/aiidateam/aiida_core/wiki/AiiDA-public-modules,-classes-and-functions) in the wiki with parts of the API that are intended to be public and as such we expose them on the second level packages. To me it only makes sense to use them as such, but as said before, indeed ultimately it is a preference

And, yes, the linter should catch those, but unfortunately for now almost all files are excluded because fixing the prospector warnings would take a lot of time. We definitely need to start uncommenting those though

Reply:
> The import robustness comes into play when importing like from `aiida.orm.nodes.data.str import Str` [...]

Right, so let's just use `from aiida.orm import Str` instead. :)

And agreed on the prospector warnings.

Thanks a lot for the chat. I don't mind each using their own preferred style.

Reply:
Fair enough, although ideally we adopt a single style and be consistent. I can certainly see your point when you just import a few classes from the module, but as soon as you start to import more than a handful, it simply seems to become messy